docs: adds an explanation page (#41)
florian-lefebvre/astro-integration-kit: docs/src/content/docs/getting-started/why.mdx

---
title: Why use Astro Integration Kit
description: Learn why you should use Astro Integration Kit to build your next Astro Integration.
---

> Astro Integration Kit is a community-driven effort to improve the power, flexibility and, most importantly, developer experience of creating Astro Integrations.
> 
> We are creating Astro Integration Kit because we want to create a robust, battle-tested suite of tools to make creating Astro integrations easier. In the hopes that one day this will become part of the core Astro framework.
> 
> We are constantly gathering feedback from the entire community, from first-time integration builders to the seasoned Astro Core team themselves.
> 
> Astro Integration Kit is currently an experimental technology that's still subject to changes as we solidify the APIs. However they have been built from techniques in use in production-level integrations across the Astro Ecosystem.
> 
> Typically Integrations require quite low-level knowledge of Astro and Vite to make successfully. So Astro Integration Kit abstracts a lot of low-level implementation into easy-to-use utilities. Things such as Vite Virtual Modules or creating Typescript Declaration files for your users.
> 
> Small utilities allow you to adhere to best practices with the minumum amount of code.
> 
> Once we've got a standard way of creating Astro Integrations we can create content and guides much more easily using well documented utilities meaning more people will be able to create more advanced integrations. 

Reviewer: ```suggestion
Astro Integration Kit is a community-driven effort to improve the power, flexibility, and, most importantly, developer experience of creating Astro Integrations.

We are creating the Astro Integration Kit because we want to create a robust, battle-tested suite of tools to make creating Astro integrations easier in the hopes that one day, this will become part of the core Astro framework.

We constantly gather feedback from the entire community, from first-time integration builders to the seasoned Astro Core team.

Astro Integration Kit is currently an experimental technology that's still subject to changes as we solidify the APIs. However, it was built from techniques used in production-level integrations across the Astro Ecosystem.

Integrations typically require quite a low-level knowledge of Astro and Vite to be made successfully. So, Astro Integration Kit abstracts a lot of low-level implementation into easy-to-use utilities. Things such as Vite Virtual Modules or creating Typescript Declaration files for your users.

Small utilities allow you to adhere to best practices with the minimum amount of code.

Once we have a standard way of creating Astro Integrations, we can create content and guides much more easily using well-documented utilities, enabling more people to create more advanced integrations.
```

We are creating the Astro Integration Kit because we want to create a robust, battle-tested suite of tools to make creating Astro integrations easier in the hopes that one day, this will become part of the core Astro framework.

We constantly gather feedback from the entire community, from first-time integration builders to the seasoned Astro Core team.

Astro Integration Kit is currently an experimental technology that's still subject to changes as we solidify the APIs. However, it was built from techniques used in production-level integrations across the Astro Ecosystem.

Integrations typically require quite a low-level knowledge of Astro and Vite to be made successfully. So, Astro Integration Kit abstracts a lot of low-level implementation into easy-to-use utilities. Things such as Vite Virtual Modules or creating Typescript Declaration files for your users.

Small utilities allow you to adhere to best practices with the minimum amount of code.

Once we have a standard way of creating Astro Integrations, we can create content and guides much more easily using well-documented utilities, enabling more people to create more advanced integrations.

:::note

> We are constantly looking for feedback and contributions from the community. Whether it's contributing code, ideas or suggestions; feel free to post an issue on the Github Repo: https://github.com/florian-lefebvre/astro-integration-kit

Reviewer 2: ```suggestion
We are constantly looking for feedback and contributions from the community. Whether it's contributing code, ideas or suggestions; feel free to create an issue on [our GitHub repository](https://github.com/florian-lefebvre/astro-integration-kit).
```

> Or get involved with the discussion on the official Astro Integrations Discord channel:  https://discord.com/channels/830184174198718474/1197638002764152843

Reviewer 2: ```suggestion
Or get involved with the discussion on the [`#dev-integrations `channel](https://discord.com/channels/830184174198718474/1197638002764152843) of the official [Astro Discord server](https://astro.build/chat)!
```
